internal/controller/warehouses/git.go

```diff
@@ -94,13 +94,22 @@ func (r *reconciler) selectCommitMeta(
 	if sub.CommitSelectionStrategy == "" {
 		sub.CommitSelectionStrategy = kargoapi.CommitSelectionStrategyNewestFromBranch
 	}
+	// for cases when ScanPaths and/or IgnorePaths filters are used we want to have 2
+	// commits cloned to be able to get actual DiffPaths of the last commit to compare
+	// against the filters
+	var cloneShallowness uint8
+	if sub.ScanPaths != nil || sub.IgnorePaths != nil {
+		cloneShallowness = 2
+	} else {
+		cloneShallowness = 1
+	}
 	repo, err := git.Clone(
 		sub.RepoURL,
 		*creds,
 		&git.CloneOptions{
 			Branch:                sub.Branch,
 			SingleBranch:          true,
-			Shallow:               true,
+			Shallow:               cloneShallowness,
```

Comment:
(If my `Depth` comment is taken into account)

```suggestion
			Depth:               &cloneDepth,
```

Reply:
agree